### ISSUE TITLE
Make it possible to mark messages as unread

### ISSUE THREAD

**Issue by alya** [zulip/zulip-mobile#5364]
In zulip/zulip#2676, we are implementing a "mark message as unread" option in the webapp.

We should also add this option to the long-press message menu in the mobile UI.

* It should be possible to mark ~~just one message, or~~ a message and all the messages below it in the current view.
* When this feature is used, the view should not mark messages as read until the user navigates away from it. We should show the usual "messages are not being marked as read" banner when this happens.

Update: Following the web app, we will only offer a "Mark as unread from here" option for now.

Related issues: #5361, #5241, #4930

**Comment by gnprice** [comment]
> * It should be possible to mark just one message, or a message and all the messages below it in the current view.

The version on web currently only supports the latter, so it's possible that's how it'll end up. Either way, we should match.

**Comment by gnprice** [comment]
> We should show the usual "messages are not being marked as read" banner when this happens.

We don't currently have a banner like this. We discussed a bit and concluded this issue will be fine without it -- when you've just marked something as unread, the fact that the screen doesn't immediately mark things read again should be intuitive enough.

We probably should have a banner to this effect for the sake of search and @-mention views. (Should file a separate issue for that.) When we do, it'd make sense to use it here too.

**Comment by gnprice** [comment]
I looked into this and found that the existing server API doesn't provide a clean way to reliably implement the feature. @timabbott then found that in fact the web implementation has a bug here. Other aspects of the web app should make the bug somewhat uncommon to trigger, but if we were to do the same thing in mobile as in web, the same bug would show up fairly often.

So I think we're going to add something to the API to support this, rather than try to support it on mobile with the existing API.

See chat thread:
https://chat.zulip.org/#narrow/stream/378-api-design/topic/mark-as-unread.20request/near/1431167

**Comment by gnprice** [comment]
https://github.com/zulip/zulip/issues/22893 tracks that API change mentioned above, which this is blocking on.

**Comment by timabbott** [comment]
As noted in https://github.com/zulip/zulip/issues/22893#issuecomment-1255663659, we've been able to figure out how to unblock merging the feature in the webapp without that API being ready quite yet (basically, we don't offer the option in situations where it's behavior would be incorrect). That same strategy could be an option for mobile to get the feature mostly working in a beta channel to get UI feedback / work out any bugs, but I expect you'll still want to wait for @andersk to finish https://github.com/zulip/zulip/issues/22893 before releasing the feature.
